### ISSUE TITLE
Getting error "twin.macro: Unexpected token, expected " using twin.macro and styled components with vite

### ISSUE THREAD

**Issue by dpovey** [ben-rogerson/twin.examples#127]
I am tearing my hair out with this. I have gone to a fair bit of trouble to make my setup match the one in [vite-styled-components-typescript](https://github.com/ben-rogerson/twin.examples/blob/master/vite-styled-components-typescript/)

I have:
1. A single Button component that is identical to the one in the example
2. package.json is same as the example save for the "name" field
3. Vanilla tailwind.config.cjs with just color overrides
4. Identical tsconfig.json with tsconfig.node.json
5. Vite config, slight difference since I want to build a library, it looks like:

```
import path from 'path'
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  optimizeDeps: {
    esbuildOptions: {
      target: 'es2020',
    },
  },
  resolve: {
    alias: {
      '@components': path.resolve(__dirname, "./src/components"),
    }
  },
  build: {
    lib: {
      entry: 'src/index.ts',
      name: 'orchard',
    },
  },
  esbuild: {
    // https://github.com/vitejs/vite/issues/8644#issuecomment-1159308803
    logOverride: { 'this-is-undefined-in-esm': 'silent' },
  },
  plugins: [
    react({
      babel: {
        plugins: ['babel-plugin-macros', 'babel-plugin-styled-components'],
      },
    }),
  ],
  define: {
    'process.env': {},
  },
  test: {
    environment: 'jsdom',
  },
})
```

I made more or less that change to the example and it still worked fine.

When I run a build it fails consistently with:

```
$  yarn build
vite v4.4.7 building for production...
✓ 1 modules transformed.
✓ built in 307ms
[vite:react-babel] .../src/components/Button.tsx: twin.macro: Unexpected token, expected "," (17:17) Learn more: https://www.npmjs.com/package/twin.macro
file: .../src/components/Button.tsx:17:17
error during build:
SyntaxError: .../src/components/Button.tsx: twin.macro: Unexpected token, expected "," (17:17) Learn more: https://www.npmjs.com/package/twin.macro
    at unexpected (.../node_modules/sucrase/dist/parser/traverser/util.js:99:15)
    at expect (.../node_modules/sucrase/dist/parser/traverser/util.js:86:5)
    at parseObj (.../node_modules/sucrase/dist/parser/traverser/expression.js:759:20)
    at parseExprAtom (.../node_modules/sucrase/dist/parser/traverser/expression.js:553:7)
    at parseExprSubscripts (.../node_modules/sucrase/dist/parser/traverser/expression.js:276:20)
    at parseMaybeUnary (.../node_modules/sucrase/dist/parser/traverser/expression.js:257:20)
    at parseExprOps (.../node_modules/sucrase/dist/parser/traverser/expression.js:185:20)
    at parseMaybeConditional (.../node_modules/sucrase/dist/parser/traverser/expression.js:157:20)
    at baseParseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:142:20)
    at tsParseMaybeAssignWithoutJSX (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1580:45)
    at tsParseMaybeAssign (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1548:12)
    at parseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:121:43)
    at parseObjectProperty (.../node_modules/sucrase/dist/parser/traverser/expression.js:851:7)
    at parseObjPropValue (.../node_modules/sucrase/dist/parser/traverser/expression.js:892:5)
    at parseObj (.../node_modules/sucrase/dist/parser/traverser/expression.js:808:5)
    at parseExprAtom (.../node_modules/sucrase/dist/parser/traverser/expression.js:553:7)
    at parseExprSubscripts (.../node_modules/sucrase/dist/parser/traverser/expression.js:276:20)
    at parseMaybeUnary (.../node_modules/sucrase/dist/parser/traverser/expression.js:257:20)
    at parseExprOps (.../node_modules/sucrase/dist/parser/traverser/expression.js:185:20)
    at parseMaybeConditional (.../node_modules/sucrase/dist/parser/traverser/expression.js:157:20)
    at baseParseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:142:20)
    at tsParseMaybeAssignWithoutJSX (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1580:45)
    at tsParseMaybeAssign (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1548:12)
    at parseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:121:43)
    at parseObjectProperty (.../node_modules/sucrase/dist/parser/traverser/expression.js:851:7)
    at parseObjPropValue (.../node_modules/sucrase/dist/parser/traverser/expression.js:892:5)
    at parseObj (.../node_modules/sucrase/dist/parser/traverser/expression.js:808:5)
    at parseExprAtom (.../node_modules/sucrase/dist/parser/traverser/expression.js:553:7)
    at parseExprSubscripts (.../node_modules/sucrase/dist/parser/traverser/expression.js:276:20)
    at parseMaybeUnary (.../node_modules/sucrase/dist/parser/traverser/expression.js:257:20)
    at parseExprOps (.../node_modules/sucrase/dist/parser/traverser/expression.js:185:20)
    at parseMaybeConditional (.../node_modules/sucrase/dist/parser/traverser/expression.js:157:20)
    at baseParseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:142:20)
    at tsParseMaybeAssignWithoutJSX (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1580:45)
    at tsParseMaybeAssign (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1548:12)
    at parseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:121:43)
    at baseParseMaybeAssign (.../node_modules/sucrase/dist/parser/traverser/expression.js:148:5)
    at tsParseMaybeAssignWithoutJSX (.../node_modules/sucrase/dist/parser/plugins/typescript.js:1580:45)
```

I'll add that I got this error consistently also with a more vanilla build that was just babel. 

Lastly node version was v16 but I tried it with v20 and same error.

Any advice you can give?

**Comment by ben-rogerson** [comment]
Hey there, sorry to hear of the issues here.

I've pasted in your custom vite config and seeing a bunch of errors (I've tried updating Vite/restarting TS/etc).

I'm also seeing a reference in the config to a file that doesn't exist in the twin example: `` entry: 'src/index.ts', ``

![image](https://github.com/ben-rogerson/twin.examples/assets/21288568/fd8c9d1e-f1d6-4251-919f-381835919a9f)

**Comment by dpovey** [comment]
Thanks for following up - I was trying to get a minimal example working with vite, vitest, styled components, storybook, typescript and the twin.macro.   I think I started with storybook example and then tried to add the other things and got stuck here.

I ended up blowing that away and then starting from scratch from the vite-styled-components-typescript example and then adding the storybook and vitest stuff in and it's now working.  Thought the error message might give you a hint, but let's just put this down to some random thing I did wrong and close it off :-). Thanks for the response!
